poetry.lock

```diff
@@ -247,6 +247,41 @@ docs = ["furo", "myst-parser", "sphinx", "sphinx-notfound-page", "sphinxcontrib-
 tests = ["attrs[tests-no-zope]", "zope-interface"]
 tests-no-zope = ["cloudpickle", "hypothesis", "mypy (>=1.1.1)", "pympler", "pytest (>=4.3.0)", "pytest-mypy-plugins", "pytest-xdist[psutil]"]
 
+[[package]]
+name = "bcrypt"
+version = "4.0.1"
+description = "Modern password hashing for your software and your servers"
+category = "main"
+optional = false
+python-versions = ">=3.6"
```

Comment:
For my own learning, where did you get these configs from? Is this just boilerplate code you copied over?

Reply:
Are you familiar with frontend tooling like `npm` or `yarn`? This poetry.lock file is like a `package-lock.json` or a `yarn.lock` -- it's generated by the package manager to specify exactly which package versions are used in the dependency tree. We define our dependencies in [pyproject.toml](https://github.com/Glitched/ReservAI/blob/14a66331ad350bc0f5089c8fe0cd5bdc17c67e17/pyproject.toml#L9). This file is generated from that list by [Poetry](https://python-poetry.org/) which is essentially a yarn-style package manager instead of the usual `pip`.

Reply:
Ah gotcha. Yes, I'm familiar with both `npm` and `yarn`. Poetry is new to me. Today I learned 🤓